app/controllers/declaration/PreviousDocumentsController.scala

```diff
@@ -85,9 +85,11 @@ class PreviousDocumentsController @Inject()(
             formWithErrors => Future.successful(BadRequest(previousDocumentsPage(formWithErrors, cache.documents))),
             updatedCache =>
               if (updatedCache != cache.documents)
-                customsCacheService
-                  .cache[PreviousDocumentsData](cacheId, formId, PreviousDocumentsData(updatedCache))
-                  .map(_ => Redirect(controllers.declaration.routes.ItemsSummaryController.displayPage()))
+                for {
```

Comment:
Couldn't you call your updateCache method here rather than have the for comprehension again?

Reply:
Crap.... completely lost that one!!!!

Reply:
Yes I should!

Reply:
This has been addressed